go-kit/metrics/provider/librato/uniquecounter.go

```diff
@@ -0,0 +1,75 @@
+/* Copyright (c) 2018 Salesforce
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root  or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+package librato
+
+import (
+	"sync"
+
+	hll "github.com/axiomhq/hyperloglog"
```

Comment:
Does this hyperloglog implementation do what we think it does? I read through the docs and it says that it is "An improved version of HyperLogLog for the count-distinct problem, approximating the number of distinct elements in a multiset using 20-50% less space than other usual HyperLogLog implementations."

Given that the cardinality of a multiset is different (sum of the multiplicities of the multiset's members) than the cardinality of a set (count of the elements in the set) I am confused.

Reply:
Hmm. The demo seems to indicate it does.

Reply:
Given the way that HLL works, I don't see how it actually can estimate the "sum of the multiplicities" as it uses a set of hashes that are deterministic. It's been a while since I've read any of the literature around this, though, so maybe I'm wrong.